File: SMeatSocialNetwork/SMeat.MODELS/Models/WorkPlace.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace SMeat.MODELS.Models
{
    public class Workplace
    {
        public Workplace()
        {
            Id = Guid.NewGuid().ToString();
        }

        [Key]
        public string Id { get; set; }

        public string Position { get; set; }

        [ForeignKey("Location")]
        public string LocationId { get; set; }

        public virtual Location Location { get; set; }

        public string CompanyName { get; set; }

        //Navigation properties
        public virtual List<User> Users { get; set; }
    }
}

File: SMeatSocialNetwork/SMeat.WEB/src/app/_models/workplace.ts
export class WorkPlace {
  id: string;
  position: string;
  locationId: string;
  location: Location;
  companyName: string;

  constructor(id?: string, position?: string, locationId?: string, location?: Location, companyName?: string) {
    this.id = id;
    this.position = position;
    this.locationId = locationId;
    this.location = location;
    this.companyName = companyName;
  }
}
